### PR TITLE
[API] [UBS_Tariffs_Courier] The system create new courier in "Тарифи" with invalid data

### PR DESCRIPTION
**Pull request by ABbondar**
# GreenCityUBS PR
https://github.com/ita-social-projects/GreenCity/issues/4801

## Summary Of Changes :fire:
Fix bug with invalid data in CreateCourierDto

## Issue Link :clipboard:
_&lt;Link to the issue&gt;_

## Added
Added regexp patterns on CreateCourierDto "nameEn" and "nameEn" fields
Added tests for regexp patterns

## Changed
* _&lt;detail item of what was changed&gt;_
* _&lt;describe functionality that was changed&gt;_

## Deleted
* _&lt;detail item of what was removed&gt;_

## How to test :clipboard:
It could be tested using swagger. Using SuperAdminController "/createCourier" endpoint (method=POST)

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [ ] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

**Review by MaksymGolik — APPROVED** [review]

**Review by LiliaMokhnatska — APPROVED** [review]

**Review by juseti — APPROVED** [review]

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java [hunk 1]
```diff
@@ -12,8 +13,12 @@
 @ToString
 public class CreateCourierDto {
     @NotNull
+    @Pattern(regexp = "[A-Za-zА0-9'\\s]{1,30}",
+        message = "use English letters, no longer than 30 symbols")
     private String nameEn;
 
     @NotNull
+    @Pattern(regexp = "[ЁёІіЇїҐґЄєА-Яа-яA0-9'\\s]{1,30}",
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: please add tests for these cases `ЁёІіЇїҐґЄє`

File: service-api/src/test/java/greencity/dto/courier/CreateCourierDtoTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class CreateCourierDtoTest {
```
Comment (RomaMocherniuk) [inline service-api/src/test/java/greencity/dto/courier/CreateCourierDtoTest.java:17]: please add tests if `nameUk` and `nameEn` contain `whitespace character`

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java (hunk 1 above)
Comment (RomaMocherniuk) [reply service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: is the case when the name starts from a number or not a capital letter valid?

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java [hunk 3]
```diff
@@ -12,8 +13,12 @@
 @ToString
 public class CreateCourierDto {
     @NotNull
+    @Pattern(regexp = "[A-Za-zА0-9'\\s]{1,30}",
+        message = "use English letters, no longer than 30 symbols")
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: Same here
is the case when the name starts from a number or not a capital letter valid?

Comment (RomaMocherniuk) [reply service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: should we add to the message that the name could contain numbers and whitespaces

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java [hunk 4]
```diff
@@ -12,8 +13,12 @@
...
     private String nameEn;
 
     @NotNull
+    @Pattern(regexp = "[ЁёІіЇїҐґЄєА-Яа-яA0-9'\\s]{1,30}",
+        message = "use Ukrainian letters, no longer than 30 symbols")
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: same here

**Comment by RomaMocherniuk** [comment]
It seems like PR name should be changed
`[API] [UBS_Tariffs_Courier] The system create new courier in "Тарифи" with invalid data #4801`

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java (hunk 4 above)
Comment (ABbondar) [reply service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: message added

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java (hunk 3 above)
Comment (ABbondar) [reply service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: fixed

File: service-api/src/test/java/greencity/dto/courier/CreateCourierDtoTest.java (hunk 2 above)
Comment (ABbondar) [reply service-api/src/test/java/greencity/dto/courier/CreateCourierDtoTest.java:17]: added

File: service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java (hunk 1 above)
Comment (ABbondar) [reply service-api/src/main/java/greencity/dto/courier/CreateCourierDto.java]: tests added, added into regex validation for the names, which starts from a number or not a capital letter

**Review by RomaMocherniuk — APPROVED** [review]
